packages/block-editor/src/components/rich-text/use-should-disable-editing.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { store as blocksStore } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { BLOCK_BINDINGS_ALLOWED_BLOCKS } from '../../hooks/use-bindings-attributes';
+import { useBlockEditContext } from '../block-edit';
+import { store as blockEditorStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+export function useShouldDisableEditing( attributeName ) {
+	const { clientId, name: blockName } = useBlockEditContext();
+	return useSelect(
+		( select ) => {
+			if (
+				! attributeName ||
+				! BLOCK_BINDINGS_ALLOWED_BLOCKS[ blockName ]?.includes(
+					attributeName
+				)
+			) {
+				return false;
+			}
+			const blockBindings =
+				select( blockEditorStore ).getBlockAttributes( clientId )
```

Comment:
This is a performance problem because we're adding another subscription to block attributes to every block (at least paragraph and all other blocks with binding support in this case). I used context in #59320.